src/Persistence/Util/ClassUtils.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Doctrine\Persistence\Util;
```

Comment:
Let's not encourage "utils". Such namespaces and classes usually end up being piles of unrelated functionality. Also, let's copy exactly what you need, nothing more. Let's not create stuff unless we know it's actually needed. This is related to proxies, so put it in a `Proxy` namespace?